magento/pwa-studio
# [PWA-1128] fix(buildpack): Stop apicache from overriding good cache headers.

## Description

Fixes a bug in `addImageOptMiddleware` where when attached, the middleware breaks all cache functionality but its own. The middleware was applying to every request, rather than every image request, and the `apicache` module was setting `Cache-Control: no-cache no-store must-revalidate` on every response it didn't recognize as a valid image.

- Added a route pattern to the `app.use()` call in `addImageOptMiddleware`, to narrow down what requests the `apicache` middleware handles to the intended set of only image requests.
- Updated `hastily` dependency to use a new version which exports its list of supported extensions, so that the image middleware can build its route pattern based on what the currently installed image optimizer supports.
- Adjusted `addImageOptMiddleware` unit test to be less brittle.

## Related Issue





Closes PWA-1128.

Tracked as a client-reported escalation through the following tickets:
PWA-1128
MC-39132
MDVA-32861

## Acceptance



### Verification Stakeholders
@novykov 
@davemacaulay 
@dpatil-magento 

Any other stakeholders from the mentioned Support tickets

### Verification Steps
1. Run `yarn && yarn run build && yarn run stage:venia`.
2. Get the response headers of an image request by:
    a. Run this command at a shell prompt: `t(){ curl -I "$1/VeniaLogo-n77.svg" }; t <staging_server_url>` (replacing `<staging_server_url>` with your local staging server URL
    b. Or, open a browser window with dev tools open, load the staging server, and observe the cache control headers on server responses
3. Observe that this request no longer has no-cache headers.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
